prov/efa/src/efa_mr.c

```diff
@@ -874,29 +887,29 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 		/* We need to add FI_REMOTE_READ to allow for Read implemented
 		* message protocols.
 		*/
-		original_access = mr_attr->access;
-		mr_attr->access |= FI_REMOTE_READ;
```

Comment:
I think it would be better to create a new mr_attribute in this branch, like `shm_mr_attr`, and fill `shm_mr_attr.hmem_data` directly with `efa_mr.gdrcopy_handler`.

Reply:
Cool thanks. Turns out we can use `ofi_mr_update_attr` to fill `mr_attr`, and it handles ABI version checks.